cmd/agent/main.go

```diff
@@ -1,3 +1,101 @@
 package main
 
-func main() {}
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"runtime"
+	"time"
+
+	"github.com/e1m0re/grdn/internal/storage"
+)
+
+func UpdateMetrics(data *storage.MetricsState) {
```

Comment:
CHANGE REQUEST: please split your application by layers. Here is business logic.